frontend/src/Editor/Editor.jsx

```diff
@@ -144,6 +144,12 @@ class Editor extends React.Component {
     });
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.appDefinition !== this.state.appDefinition) {
```

Comment:
@arpitnath this doesn't do a deep equality check, 

```
var a = {x: { a: 2 }}
var b = {x: { a: 2 }}

a === b // returns false
```

Can we use lodash's `isEqual` here?